src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/ObjectHeader.cs

```diff
@@ -58,24 +56,21 @@ public static unsafe int ReadVolatileMemory(int* pHeader)
         public static unsafe int GetHashCode(object o)
         {
             if (o == null)
-            {
                 return 0;
-            }
 
-            fixed (byte* pRawData = &o.GetRawData())
+            fixed (MethodTable** ppMethodTable = &o.GetMethodTableRef())
```

Comment:
Pinning the object using `GetMethodTableRef` is not generally ok, but it is ok here since it is very low-level ObjectHeader code.

Reply:
Pinning method table ptr saves computing the offset to data. As long as byref is within the range of the object, it is all the same to GC as it will "align" to the containing object and pin that object.

I wish the object header belonged to the object though. We would not need pinning at all.